init.lua

```diff
@@ -33,111 +74,304 @@ local function get_int_attribute(player, key)
-local enable_sprint_particles = minetest.setting_getbool("sprint_particles") ~= false
-local armor_mod = minetest.get_modpath("3d_armor")
-
-function set_sprinting(name, sprinting)
-	local player = minetest.get_player_by_name(name)
-	local def = {}
-	-- Get player physics from 3d_armor mod
-	if armor_mod and armor and armor.def then
-		def.speed = armor.def[name].speed
-		def.jump = armor.def[name].jump
-		def.gravity = armor.def[name].gravity
+function stamina.set_sprinting(player, sprinting)
+	for _, fun in ipairs(stamina.registered_on_sprintings) do
+		local rv = fun(player, sprinting)
+		if rv == true then
+			return
+		end
 	end
 
-	def.speed = def.speed or 1
-	def.jump = def.jump or 1
-	def.gravity = def.gravity or 1
+	if player_monoids_mod then
+		if sprinting then
+			player_monoids.speed:add_change(player, 1 + settings.sprint_speed, "stamina:physics")
+			player_monoids.jump:add_change(player, 1 + settings.sprint_jump, "stamina:physics")
+		else
+			player_monoids.speed:del_change(player, "stamina:physics")
+			player_monoids.jump:del_change(player, "stamina:physics")
+		end
+	else
+		local def
+		if armor_mod then
+			-- Get player physics from 3d_armor mod
+			local name = player:get_player_name()
+			def = {
+				speed=armor.def[name].speed,
+				jump=armor.def[name].jump,
+				gravity=armor.def[name].gravity
+			}
+		else
+			def = {
+				speed=1,
+				jump=1,
+				gravity=1
+			}
+		end
 
-	if sprinting == true then
+		if sprinting then
+			def.speed = def.speed + settings.sprint_speed
+			def.jump = def.jump + settings.sprint_jump
+		end
 
-		def.speed = def.speed + SPRINT_SPEED
-		def.jump = def.jump + SPRINT_JUMP
+		player:set_physics_override(def)
 	end
 
-	player:set_physics_override({
-		speed = def.speed,
-		jump = def.jump,
-		gravity = def.gravity
-	})
+	if settings.sprint_particles and sprinting then
+		local pos = player:getpos()
+		local node = minetest.get_node({x = pos.x, y = pos.y - 1, z = pos.z})
+		if node.name ~= "air" then
+			minetest.add_particlespawner({
+				amount = 5,
+				time = 0.01,
+				minpos = {x = pos.x - 0.25, y = pos.y + 0.1, z = pos.z - 0.25},
+				maxpos = {x = pos.x + 0.25, y = pos.y + 0.1, z = pos.z + 0.25},
+				minvel = {x = -0.5, y = 1, z = -0.5},
+				maxvel = {x = 0.5, y = 2, z = 0.5},
+				minacc = {x = 0, y = -5, z = 0},
+				maxacc = {x = 0, y = -12, z = 0},
+				minexptime = 0.25,
+				maxexptime = 0.5,
+				minsize = 0.5,
+				maxsize = 1.0,
+				vertical = false,
+				collisiondetection = false,
+				texture = "default_dirt.png",
+			})
+		end
+	end
 end
+--- END SPRINTING API ---
 
 -- Time based stamina functions
+local function move_tick()
+	for _,player in ipairs(minetest.get_connected_players()) do
+		local controls = player:get_player_control()
+		local is_moving = controls.up or controls.down or controls.left or controls.right
+		local velocity = player:get_player_velocity()
+		local has_velocity = velocity.x ~= 0 or velocity.z ~= 0
```

Comment:
Why not `vector.length(velocity) > 0.05`? That would also allow some floating point errors if they're only standing.